src/main/java/pl/szczesniak/dominik/whattowatch/movies/domain/MoviesToWatchService.java

```diff
@@ -0,0 +1,58 @@
+package pl.szczesniak.dominik.whattowatch.movies.domain;
+
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import pl.szczesniak.dominik.whattowatch.movies.domain.model.MovieId;
+import pl.szczesniak.dominik.whattowatch.movies.domain.model.MovieTitle;
+import pl.szczesniak.dominik.whattowatch.movies.domain.model.exceptions.MovieDoesNotExistException;
+import pl.szczesniak.dominik.whattowatch.movies.domain.model.exceptions.UserDoesNotExistException;
+import pl.szczesniak.dominik.whattowatch.users.domain.model.UserId;
+
+import java.util.List;
+
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
+public class MoviesToWatchService {
+
+	private final MoviesRepository repository;
+	private final UserProvider userProvider;
+	private final WatchedMoviesRepository watchedRepository;
+
+	public MovieId addMovieToList(final String movieTitle, final UserId userId) {
+		userCheck(userId);
+		final Movie movie = new Movie(repository.nextMovieId(), new MovieTitle(movieTitle), userId);
+		repository.save(movie);
+		return movie.getMovieId();
+	}
+
+	public void removeMovieFromList(final MovieId movieId, final UserId userId) {
+		repository.removeMovie(movieId, userId);
+	}
+
+	public List<Movie> getList(final UserId userId) {
+		return repository.findAll(userId);
+	}
+
+	public void moveMovieToWatchedList(final MovieId movieId, final UserId userId) {
+		userCheck(userId);
+		final WatchedMovie watchedMovie = new WatchedMovie(
```

Comment:
to jest ok, ale przypomnij mi o tym to pokaże Ci jak można by to było zrefactorować żeby nie trzeba było wyciągasz z filmu wszystkich danych.

Możesz sam to też spróbować zrefactorować. Chciałbym uzyskać coś takiego:
```java
final Movie existingMovie = repository.findBy(movieId, userId).orElseThrow(() -> new MovieDoesNotExistException("Movie doesn't match userId: " + userId));
final WatchedMovie watchedMovie = existingMovie.markAsWatched();
```